File: CodeChef/JulyLongChallenge/EGRANDR.py
for i in range(int(raw_input())):
    dis = int(raw_input())
    n = map(int, raw_input().split())
    s = 0
    flag = False
    full = False
    
    for marks in n:
        if marks == 2:
            flag = True
            break
        else:
            if marks == 5:
                full = True
            s = s + marks

    if flag or not full:
        print "No"
    else:
        avg = s / len(n)
        if avg < 4:
            print "No"
        else:
            print "Yes"

File: CodeChef/JulyLongChallenge/EGRANDR.cpp
#include <iostream>

using namespace std;

int main()
{
    int t;
    cin>>t;

    while(t--)
    {
        float no;
        cin>>no;

        bool flag = false, fullscore = false;
        float sum = 0;
        for(int i = 0;i < no; i++)
        {
            int marks;
            cin>>marks;

            if(marks == 2)
            {
                flag = true;
                break;
            }
            else
            {
                if(marks == 5)
                    fullscore = true;
                sum += marks;
            }
        }

        if(flag || !fullscore)
        {
            cout<<"No\n";
        }
        else
        {
            float avg = sum / no;
            if(avg < 4)
            {
                cout<<"No\n";
            }
            else
                cout<<"Yes\n";
        }
    }
    return 0;
}